cEP-0025.md

```diff
@@ -0,0 +1,233 @@
+        for _, node in scope.items():
+            if isinstance(node, node_type):
+                yield node
+
+
+class PyFlakesASTBear(LocalBear):
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    def run(self, filename, file):
+        """
+        Generates pyflakes-enhance-AST for the input file and returns
+        deadScopes as HiddenResult
+        :return:
+            One HiddenResult containing a dictionary with keys being
+            type of scope and values being a list of scopes generated
+            from the file.
+        """
+        tree = ast.parse(''.join(file))
+        result = Checker(tree, filename=filename, withDoctest=True)
+
+        yield PyFlakesResult(self, result.deadScopes, result.messages)
+```
+
+## Demonstrating use of PyFlakesASTBear by creating a simple bear
+
+A hasstle free solution to the problem of finding all the future imports
+present in the source code.
+
+Here is a prototype for the NoFutureImportBear that uses PyFlakesASTBear:
+
+```python
+# Imports not mentioned to maintain brevity
+
+class NoFutureImportBear(LocalBear):
+    """
+    Uses pyflakes-enhance-AST to detect use of future imports
+    """
+    BEAR_DEPS = {PyFlakesASTBear}
+
+    def run(self, filename, file,
+            dependency_results=dict()
+            ):
+        for result in dependency_results.get(PyFlakesASTBear.name, []):
+            for node in result.get_nodes(result.module_scope[0],
+                                         FutureImportation):
+                lineno = result.get_node_location(node)[0]
+                corrected = self.remove_import(file, lineno, node.name)
+                yield Result.from_values(
+                    origin=self,
+                    message=('Future import %s found' % node.name),
+                    file=filename,
+                    diffs={filename: corrected},
+                    line=result.get_node_location(node)[0])
+```
+
+## Demonstrating use of PyFlakesASTBear by creating a fixes bear
+
+A hasstle free solution to the problem of removing all the future imports
+present in the source code.
+
+Here is a prototype for the NoFutureImportBear that uses PyFlakesASTBear:
+
+```python
+# Imports not mentioned to maintain brevity
+import astor
+
+class RemoveFutureImports(ast.NodeTransformer):
+    def visit_ImportFrom(self, node):
+        try:
+            if node.module == '__future__':
+                return None
+        except AttributeError:
+            pass
+        return node
+
+class NoFutureImportBear(LocalBear):
+    """
+    Uses pyflakes-enhance-AST to remove future imports
+    """
+    BEAR_DEPS = {PyFlakesASTBear}
+
+    def remove_future_imports(self, file, lineno):
+        """
+        Removes all ImportFrom nodes from the input line
+        """
+        diff = Diff(file)
+        line = file[lineno - 1]
+        tree = ast.parse(line)
```

Comment:
hmm.  three days ... no reply

Reply:
Working on it. Cases like =>
```
from __future__ import print_function, \
                        unicode_literals 
```
are a real problem. Since the lineno for both the nodes are 1, 1 and not 1, 2 which I didn't expected.